lib/array_intersection.rb

```diff
@@ -1,6 +1,22 @@
 # Returns a new array to that contains elements in the intersection of the two input arrays
-# Time complexity: ?
-# Space complexity: ?
+# Time complexity: O(n*m), where n is the number of elements in array1 and m is the number of elements in array2
+# Space complexity: O(n), where n is less than m
```

Comment:
Except n is not necessarily less than m. A more accurate statement would be: O(min(n,m)). Also, not that that's the worst case; when one array is a subset of the other. The typical case will be smaller.